packages/ui/ImageItem/ImageItem.js

```diff
@@ -123,42 +185,80 @@ const ImageItemBase = kind({
 			selected,
 			horizontal: orientation === 'horizontal',
 			vertical: orientation === 'vertical'
-		})
+		}),
+		computedProps: ({children, css, imageComponent, orientation, placeholder, src, ...rest}) => (reducedComputed({
+			isHorizntal: () => (orientation === 'horizontal'),
+			memoImage: ({isHorizntal}) => {
+				return React.useMemo(() => {
+					return (
+						<Cell
+							className={css.image}
+							component={ImageOverride}
+							imageComponent={imageComponent}
+							key="image"
+							placeholder={placeholder}
+							shrink={isHorizntal}
+							src={src}
+						/>
+					);
+					// We don't need the dependency of the `src` because it will be passed through a context.
+					// eslint-disable-next-line react-hooks/exhaustive-deps
+				}, [css.image, imageComponent, isHorizntal, placeholder]);
+			},
+			memoCaption: () => {
+				return children ? React.useMemo(() => {
+					return children;
+				}, [children]) : null;
+			},
+			memoChildren: ({memoCaption, isHorizntal}) => {
+				return memoCaption ? React.useMemo(() => {
+					return (
+						<Cell
+							// eslint-disable-next-line no-undefined
+							align={isHorizntal ? 'center' : undefined}
+							className={css.caption}
+							key="children"
+							shrink={!isHorizntal}
+						>
+							{memoCaption}
+						</Cell>
+					);
+				}, [css.caption, isHorizntal, memoCaption]) : null;
+			},
+			computedProps: ({isHorizntal, memoChildren, memoImage}) => ({isHorizntal, memoChildren, memoImage, rest})
+		}))
 	},
 
-	render: ({children, css, imageComponent, orientation, placeholder, src, ...rest}) => {
-		delete rest.selected;
+	render: ({className, computedProps: {isHorizntal, memoChildren, memoImage, rest}}) => {
+		const Component = isHorizntal ? Row : Column;
 
-		const isHorizontal = orientation === 'horizontal';
-		const Component = isHorizontal ? Row : Column;
+		delete rest.className;
+		delete rest.selected;
 
 		return (
-			<Component {...rest}>
-				<Cell
-					className={css.image}
-					component={ImageOverride}
-					imageComponent={imageComponent}
-					placeholder={placeholder}
-					shrink={isHorizontal}
-					src={src}
-				/>
-				{children ? (
-					<Cell
-						className={css.caption}
-						shrink={!isHorizontal}
-						// eslint-disable-next-line no-undefined
-						align={isHorizontal ? 'center' : undefined}
-					>
-						{children}
-					</Cell>
-				) : null}
-			</Component>
+			<MemoPropsDOMAttributesContext attr={['data-index']}>
+				{
+					React.useMemo(() => {
+						return (
+							<Component {...rest} className={className}>
```

Comment:
If the `selected` class can go to Image, we could reduce re-render by that class. If you concern.